code/WorkInProgress/nadir_antenna.dm

```diff
@@ -0,0 +1,689 @@
+						oofed_nerds += AM
+						continue
+					if(isobj(AM))
+						var/obj/O = AM
+						if(istype(O,/obj/storage/crate) || O.artifact)
+							thing2send = O
+							break //only one thing at a time!
+				for(var/nerd in oofed_nerds)
+					telefrag(nerd) //did I mention NO MOBS
+				if(thing2send && transception_array.transceive(netnumber))
+					thing2send.loc = src
+					SPAWN(1 SECOND)
+
+						if (istype(thing2send, /obj/storage/crate/biohazard/cdc))
+							QM_CDC.receive_pathogen_samples(thing2send)
+
+						else if(istype(thing2send,/obj/storage/crate) || istype(thing2send,/obj/storage/secure/crate))
+							var/sold_to_trader = FALSE
+							for (var/datum/trader/T in shippingmarket.active_traders)
+								if (T.crate_tag == thing2send.delivery_destination)
+									shippingmarket.sell_crate(thing2send, T.goods_buy)
+									sold_to_trader = TRUE
+									break
+							if(!sold_to_trader)
+								shippingmarket.sell_crate(thing2send)
+
+						else if(thing2send.artifact)
+							shippingmarket.sell_artifact(thing2send)
+
+						else //how even
+							logTheThing("debug", null, null, "Telepad attempted to send [thing2send], which is not a crate or artifact")
+
+				showswirl(src.loc)
+				use_power(200) //most cost is at the array
+				src.is_transceiving = FALSE
+		return
+
+	proc/receive_a_thing(var/netnumber,var/atom/movable/thing2get)
+		src.is_transceiving = TRUE
+		if(thing2get in shippingmarket.pending_crates)
+			shippingmarket.pending_crates.Remove(thing2get) //avoid received thing being queued into multiple pads at once
+		playsound(src.loc, "sound/effects/ship_alert_minor.ogg", 50, 0) //incoming cargo warning (stand clear)
+		SPAWN(2 SECONDS)
+			flick("neopad_activate",src)
+			SPAWN(0.4 SECONDS)
+				var/tele_obstructed = FALSE
+				var/turf/receive_turf = get_turf(src)
+				if(length(receive_turf.contents) < 10) //fail if there is excessive clutter or dense object
+					for(var/atom/movable/O in receive_turf)
+						if(istype(O,/obj))
+							if(O.density)
+								tele_obstructed = TRUE
+						if(istype(O,/mob/living/carbon/human) && prob(25))
+							telefrag(O) //get out the way
+				else
+					tele_obstructed = TRUE
+				if(!tele_obstructed && transception_array.transceive(netnumber))
+					thing2get.loc = src.loc
+					showswirl(src.loc)
+					use_power(200) //most cost is at the array
+				else
+					shippingmarket.pending_crates.Add(thing2get)
+					playsound(src.loc, "sound/machines/pod_alarm.ogg", 30, 0)
+					src.visible_message("<span class='alert'><B>[src]</B> emits an [tele_obstructed ? "obstruction" : "array status"] warning.</span>")
+				src.is_transceiving = FALSE
+
+		return
+
+	///Standing on the pad while it's trying to transport cargo is an extremely dumb idea, prepare to get owned
+	proc/telefrag(var/mob/living/carbon/human/M)
+		var/dethflavor = pick("suddenly vanishes","tears off in the teleport stream","disappears in a flash","violently disintegrates")
+
+		switch(rand(1,4))
```

Comment:
The possibility of whooshing on a limb if you're already missing it is reasonably desirable to discourage use of the transception pad as a torture chamber; to avoid the weird behavior when rolling a limb that's already missing, I'm adding a check that makes the stun and scream only go off if you actually got a limb ripped off.